### PR TITLE
Add Typing for EmailField with Support for String Set and Get Types

### LINKED ISSUE #2212: `django.contrib.auth.models.User.email` allows assignment of an integer
# Bug report

`django.contrib.auth.models.User.email` allows assignment of an integer. It's not just a problem for the `User` model but for all instances of `EmailField`.

## What's wrong

```python
from django.contrib.auth.models import User
user = User()
user.email = 1
```

## How is that should be

There should be an "Incompatible types in assignment ..."

## System information

- OS:
- `python` version: `3.12.3`
- `django` version: `5.0.6`
- `mypy` version: `1.10.0`
- `django-stubs` version: `master`
- `django-stubs-ext` version: `master`

### PR DESCRIPTION
**Pull request by ashm-dev**
# I have made things!

## Related issues
Closes #2212

### CODE REVIEW

File: django-stubs/db/models/fields/__init__.pyi [hunk 1]
```diff
@@ -356,7 +356,9 @@ class SlugField(CharField[_ST, _GT]):
         allow_unicode: bool = ...,
     ) -> None: ...
 
-class EmailField(CharField[_ST, _GT]): ...
+class EmailField(CharField[_ST, _GT]):
+    _pyi_private_set_type: str | Combinable
+    _pyi_private_get_type: str
```
Comment (sobolevn) [inline django-stubs/db/models/fields/__init__.pyi]:
```suggestion
    _pyi_private_set_type: str | Combinable
```

Since `CharField` already has `_pyi_private_get_type: str` set.

**Comment by ashm-dev** [comment]
@sobolevn Removed the duplication, thanks for your help!)

File: django-stubs/db/models/sql/compiler.pyi [hunk 2]
```diff
@@ -13,9 +13,9 @@ from django.db.models.sql.subqueries import AggregateQuery, DeleteQuery, InsertQ
 from django.utils.functional import cached_property
 from typing_extensions import TypeAlias
 
-_ParamT: TypeAlias = str | int
+_ParamT: TypeAlias = Any
```
Comment (sobolevn) [inline django-stubs/db/models/sql/compiler.pyi]: This does not look related to EmailFiled change

**Review by sobolevn — CHANGES_REQUESTED** [review]

File: django-stubs/db/models/sql/compiler.pyi (hunk 2 above)
Comment (ashm-dev) [reply django-stubs/db/models/sql/compiler.pyi]: Data from another PR accidentally got here, I'm sorry)

**Review by sobolevn — APPROVED** [review]
Thanks! 👍
